sympy/physics/mechanics/_pathway.py

```diff
@@ -0,0 +1,273 @@
+
+    To construct a pathway, two points are required to be passed to the
+    ``origin`` and ``insertion`` parameters.
+
+    >>> from sympy.physics.mechanics import Point
+    >>> origin = Point('pO')
+    >>> insertion = Point('pI')
+    >>> linear_pathway = LinearPathway(origin=origin, insertion=insertion)
+    >>> linear_pathway
+    LinearPathway(origin=pO, insertion=pI)
+
+    The pathway create above isn't very interesting without the positions and
+    velocities of its origin and insertion points being described. Without this
+    its not possible to describe how the pathway moves, i.e. its length or its
+    shortening velocity.
+
+    >>> from sympy.physics.mechanics import ReferenceFrame
+    >>> from sympy.physics.vector import dynamicsymbols
+    >>> N = ReferenceFrame('N')
+    >>> q = dynamicsymbols('q')
+    >>> insertion.set_pos(origin, q * N.x)
+    >>> insertion.pos_from(origin)
+    q(t)*N.x
+
+    A pathway's length can be accessed via its ``length`` attribute.
+
+    >>> linear_pathway.length
+    sqrt(q(t)**2)
+
+    Note how what appears to be an overly-complex expression is returned. This
+    is actually required as it ensures that a pathway's length is always
+    positive.
+
+    A pathway's shortening velocity can be accessed similarly via its
+    ``shortening_velocity`` attribute.
+
+    >>> linear_pathway.shortening_velocity
+    -q(t)*Derivative(q(t), t)/sqrt(q(t)**2)
+
+    Parameters
+    ==========
+
+    origin : Point
+        The ``Point`` from which the pathway originates.
+    insertion : Point
+        The ``Point`` into which the pathway inserts.
+
+    """
+
+    def __init__(
+        self,
+        *,
+        origin: Point,
+        insertion: Point,
+    ) -> None:
+        """Initializer for ``LinearPathway``.
+
+        Parameters
+        ==========
+
+        origin : Point
+            The ``Point`` from which the pathway originates.
+        insertion : Point
+            The ``Point`` into which the pathway inserts.
+
+        """
+        super().__init__(
+            origin=origin,
+            insertion=insertion,
+        )
+
+    def _true_length(self) -> Expr:
+        """Exact analytical expression for the pathway's length."""
+        length = self.insertion.pos_from(self.origin).magnitude()
+        return length
+
+    def _true_shortening_velocity(self) -> Expr:
+        """Exact analytical expression for the pathway's shortening velocity."""
+        relative_position = self.insertion.pos_from(self.origin)
+        if not relative_position:
+            return S.Zero
+        t = dynamicsymbols._t  # type: ignore
+        # A reference frame is needed to differentiate ``relative_position`` to
+        # ``relative_velocity`` so choose the first ``ReferenceFrame`` that
+        # ``relative_position`` is defined using.
+        frame = relative_position.args[0][1]
+        relative_velocity = relative_position.diff(t, frame)
```

Comment:
I'm a bit in doubt on this solution. Mainly because it might be preferable to set the shortening velocity yourself as a user in terms of generalized speeds (that is way more efficient). Not sure what a nice way would be to specify this. I would also have to really dive in whether there are no obvious problems you may get by using this method.

Reply:
May `self.inertion.vel(frame) - self.origin.vel(frame)` actually be better? Have to say that you can still not really as a user make this more efficient, because you have no control over the frame that is being used.

Reply:
I agree it would be nice if it was possible to define the shortening velocity is a more efficient way. However, if the user sets it then there won't be an guarantee on correctness. Are we happy with that?

I suggest we leave as is for the time being and address issues as we encounter them. We have the protection of being able to make breaking API changes in the short term.

Reply:
> May `self.inertion.vel(frame) - self.origin.vel(frame)` actually be better?

I remember trying both options previously and us deciding that it didn't make a different. Do you have any reason why one might be preferred over the other?

Reply:
> there won't be an guarantee on correctness. Are we happy with that?

Are we ever sure a user did specify stuff correct. It is mainly for the option to have the flexibility from an advance user perspective.

> I remember trying both options previously and us deciding that it didn't make a different. Do you have any reason why one might be preferred over the other?

`diff` always computes it based by taking the derivative, while `.vel` can use the stored velocity (this would also make that this method would kind of start using cache, though bad cache). Using the stored velocity can be better as it may be expressed in the generalized speeds. The disadvantage is that you may end up with something more complex, because both points their velocities are possibly computed with respect to another arbitrary point, which doesn't even necessarily have to be the same.

Reply:
Okay, I suggest that we perhaps leave this but open an issue about it to remind ourselves that this needs investigating as the functionality becomes more fully-fledged?

Reply:
Similar concern to Timo's came to me. Maybe open an issue so we don't forget about it. I'm also not sure selecting the frame like you have guarantees things are correct.